pkg/cloud/gcp/actuators/machine/reconciler_test.go

```diff
@@ -330,6 +330,62 @@ func TestCreate(t *testing.T) {
 				}
 			},
 		},
+		{
+			name: "Restart policy defaults to Always when not set",
+			validateInstance: func(t *testing.T, instance *compute.Instance) {
+				if instance.Scheduling != nil && instance.Scheduling.AutomaticRestart != nil && *instance.Scheduling.AutomaticRestart != true {
+					t.Errorf("Expected to find instance automatic restart set to true")
+				}
+			},
+		},
+		{
+			name: "Restart policy is set to Always when set",
+			providerSpec: &machinev1.GCPMachineProviderSpec{
+				RestartPolicy: machinev1.RestartPolicyAlways,
+			},
+			validateInstance: func(t *testing.T, instance *compute.Instance) {
+				if instance.Scheduling != nil && instance.Scheduling.AutomaticRestart != nil && *instance.Scheduling.AutomaticRestart != true {
+					t.Errorf("Expected to find instance automatic restart set to true")
+				}
+			},
+		},
+		{
+			name: "Restart policy is set to Never when set",
+			providerSpec: &machinev1.GCPMachineProviderSpec{
+				RestartPolicy: machinev1.RestartPolicyNever,
+			},
+			validateInstance: func(t *testing.T, instance *compute.Instance) {
+				if instance.Scheduling != nil && instance.Scheduling.AutomaticRestart != nil && *instance.Scheduling.AutomaticRestart != false {
+					t.Errorf("Expected to find instance automatic restart set to true")
+				}
+			},
+		},
+		{
+			name: "Restart policy is set to Never when using preemptible instances",
```

Comment:
this test case might be worth dropping since it's not necessarily testing the logic in our function, but it does confirm the default behavior coming out of the gcp sdk. happy to remove if folks don't think we should include it.